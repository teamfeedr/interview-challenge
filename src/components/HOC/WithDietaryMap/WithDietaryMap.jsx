import React from 'react';
import PropTypes from 'prop-types';

const getDietaryMap = ({ items }) => items.reduce((dietaryMap, item) => {
    item.dietaries && item.dietaries.forEach((diet) => {
        if (dietaryMap.has(diet)) {
            dietaryMap.set(diet, dietaryMap.get(diet) + 1)
        }
        else {
            dietaryMap.set(diet, 1);
        }
    })

    return dietaryMap;
}, new Map());

const WithDietaryMap = ({ items, WrappedComponent, ...childProps }) => {
    const dietaryMap = getDietaryMap({ items });

    return (<WrappedComponent dietaryMap={dietaryMap} selectedCount={items.length} {...childProps} />);
}

WithDietaryMap.propTypes = {
    WrappedComponent: PropTypes.func.isRequired,
    items: PropTypes.arrayOf(PropTypes.object).isRequired
}

export default WithDietaryMap;