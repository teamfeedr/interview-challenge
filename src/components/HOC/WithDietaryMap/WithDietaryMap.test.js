import React from "react";
import { shallow } from "enzyme";

import WithDietaryMap from './WithDietaryMap';
import items from '../../../items';

describe('the WithDietaryMap HOC', () => {
    it('renders a wrapped component with a dietary Map', () => {
        const TestComponent = () => {return (<div/>)};
        
        const rendered = shallow(<WithDietaryMap items={items} WrappedComponent={TestComponent}/>)

        expect(rendered).toMatchSnapshot();
    })
})