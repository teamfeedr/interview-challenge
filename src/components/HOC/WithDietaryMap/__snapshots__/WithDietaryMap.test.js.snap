// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the WithDietaryMap HOC renders a wrapped component with a dietary Map 1`] = `
<TestComponent
  dietaryMap={
    Map {
      "v" => 8,
      "ve" => 7,
      "df" => 15,
      "gf" => 19,
      "n!" => 7,
      "rsf" => 4,
    }
  }
  selectedCount={20}
/>
`;
